### ISSUE TITLE
[@types/next] Wrong query types?

### ISSUE THREAD

**Issue by herrstucki** [DefinitelyTyped/DefinitelyTyped#26843]
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @resir014

@resir014 May I ask why you reverted the query types in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25711/commits/512782a734254850dc31005b359cdf3f5239aa75 ? For the record, I think both versions are not quite correct … shouldn't it be like this?

```ts
query: { [key: string]: string | string[] | undefined }
```

AFAIK, the query can contain arrays but is never parsed into numbers/booleans. Also, any key can be undefined, so I think the type should need to account for that.

**Comment by resir014** [comment]
I think @brikou can shed some light to this, I just done that based on his review. But yeah, I agree that `any` is better for queries.

**Comment by herrstucki** [comment]
I don't think it should be `any`? A query property can only be `string | string[] | undefined`, no?

Is the current type maybe because `query` is asymmetric (i.e. you can _set_ a number but you'll always _get_ a string)?

**Comment by resir014** [comment]
Ah yes my bad, you're right.

I haven't had time to touch the typings so far (busy weeks at work!), but I can put together a fix tomorrow, alongside #26562.
